logger/syslogd/syslogd.go

```diff
@@ -97,7 +101,7 @@ func (h *handler) mainLoop() {
 }
 
 // Listen starts a new syslog server which runs until it receives a signal.
-func Listen(signalChan chan os.Signal, cleanupDone chan bool, bindAddr string) {
+func Listen(exitChan, cleanupDone chan bool, drainChan chan string, bindAddr string) {
```

Comment:
one possible solution is to have a signal notifier inside `syslogd.Listen` and one in main for the etcd watcher but I believe that it's better to handle all signals from main. I'm not opposed to have one channel per signal.